packages/ui-extensions/src/surfaces/point-of-sale/components/CameraScanner/CameraScanner.ts

```diff
@@ -9,21 +9,43 @@ import {BannerVariant} from '../Banner/Banner';
 export interface CameraScannerBannerProps {
   /**
    * The title of the banner.
+   * @deprecated CameraScannerBannerProps has been deprecated. Use the CameraScannerNextProps instead.
    */
   title: string;
 
   /**
    * The appearance of the banner.
+   * @deprecated CameraScannerBannerProps has been deprecated. Use the CameraScannerNextProps instead.
    */
   variant: BannerVariant;
+
   /**
    * The visibility state of the banner.
+   * @deprecated CameraScannerBannerProps has been deprecated. Use the CameraScannerNextProps instead.
    */
   visible: boolean;
 }
 
+/**
+ * @property message - The error message to show in the banner.
+ */
+
+interface CameraScannerError {
+  message: string;
+}
+
+/**
+ * @property scannerState - the state of scanning the barcode.
+ * @property overlayText - the scanner overlay default text on top of the scanner window.
+ */
+
 export interface CameraScannerProps {
+  /**
+   * @deprecated bannerProps has been deprecated. Use scannerState and overlayText props instead.
+   */
   bannerProps?: CameraScannerBannerProps;
+  scannerState?: CameraScannerError | 'SUCCESS';
+  overlayText?: string;
```

Comment:
Would you explain how the extension developer should work with the proposed props to do the following workflow?

1. CameraScanner opens `scannerState: undefined`
2. Some scan data comes in and results in an error `scannerState: {message: 'Error'}`
3. The user dismisses the banner (is there any feedback here?)
4. The scanner goes back into a scanning mode
5. Another scan comes in and results in an error (scannerState is probably still in error, how does the extension resurface the banner? does it have to explicitly set the scannerState `undefined` and then back to error?)

One option would be to add an `onDismiss` function to the `CameraScannerError` interface so that the extension can be notified that the banner got dismissed.

Does dismissal of the banner make the scanner transition state from Error to Scanning? You mentioned attaching a video to the PR but I did not see it. Maybe that would help me understand.

If there are three possible states and both the extension and the scanner itself can both cause state transitions, enhancing the ScannerAPI might be a better idea than modifying these props. If the ScannerAPI provided a subscribable to the state of the scanner as well as a function to transition the state, would that solve the problem?

@js-goupil for sanity check on the above

Reply:
It will more or less something like that 
```tsx
const CameraScannerNext = ({testID, scannerState, overlayText}: TestProps & CameraScannerProps) => {
  const {setScannerData} = useContext(ScannerContext);

  const [scannerBanner, setBannerData] = useState<BannerProps | undefined>(undefined);

  const {send, overlayReticleType, overlayTextType} = useScannerOverlayState();

  const onPOSBarcodeScanned = (barcode: ScanResult) => {
    setScannerData?.({data: barcode.data, source: 'camera'});
    if (barcode.data && scannerState=== 'SUCCESS') {
      send.onSuccess();
      setBannerData(undefined);
    } else if (barcode.data && CameraScannerError) {
      send.onError({setToBlank: true, message: null});
      setBannerData({
        title: bannerProps.title,
        status: 'critical',
        action: {
          label: 'Dismiss',
          onPress: () => {
            setBannerData(undefined);
            send.onReset();
          },
        },
      });
    } else {
      setBannerData({
        title: bannerProps.title,
        status: toBannerType(bannerProps.variant),
        action: {
          label: 'Dismiss',
          onPress: () => {
            setBannerData(undefined);
            send.onReset();
          },
        },
      });
    }
  };

  return (
    <POSVirtualBarcodeScannerNext
      onBarcodeScanned={onPOSBarcodeScanned}
      testID={testID}
      flipButtonPosition="bottom"
      overlayReticleType={overlayReticleType}
      overlayTextType={overlayTextType}
    >
      <ScannerOverlayBanner>
        {scannerBanner ? <Banner title={bannerProps.title} status={toBannerType(bannerProps.variant)} /> : null}
      </ScannerOverlayBanner>
    </POSVirtualBarcodeScannerNext>
  );
};
```

Reply:
https://github.com/user-attachments/assets/b4d38937-5de1-4563-ba28-b6de2d14400f

Reply:
example how it is used in POS 
https://github.com/Shopify/pos-next-react-native/blob/3de88a8fbad84b0644148ae2486e4f759ca38d01/src/screens/Orders/OrderDetails/FulfillmentCameraScannerNext.tsx

Reply:
there is a draft PR on the POS side 
https://github.com/Shopify/pos-next-react-native/pull/40572/files